relayer/relayer.go

```diff
@@ -189,9 +217,8 @@ func (r *Relayer) processMissedBlocks(
 			)
 		}
 		return nil
-	}
-	if errors.Is(err, database.ErrChainNotFound) || errors.Is(err, database.ErrKeyNotFound) {
-		// Otherwise, latestProcessedBlock is nil, so we instead store the latest block height.
+	} else {
+		// Otherwise, do not process any historical blocks, and store the latest block height
```

Comment:
I thought in this case we would log a warning and process starting from block 0?

Reply:
The way this function is called, we should never hit this case since `startBlockHeight` will never be `nil`.

Looking at it again though, it'd probably be cleaner to have this function require `startBlockHeight` to be non-nil, and remove this branch altogether.